build_script.py

```diff
@@ -184,40 +185,60 @@ def main(args=sys.argv[1:]):
+        description=textwrap.dedent("""
+            Build, test, and install XCTest.
+
+            WARNING: This script is only meant to be used on Linux host
+            machines, and in general should not be invoked directly. The
+            recommended way to build and test XCTest is via the Swift build
+            script. See this project's README for details.
+
+            The Swift build script invokes this %(prog)s script to build,
+            test, and install this project on Linux. Assuming you are using a
+            Linux host machine, you may invoke it in the same way to build this
+            project directly. For example, If your install of Swift is located
+            at "/swift" and you wish to install XCTest into that same location,
+            here is a sample invocation of the build script:
+
+            $ %(prog)s \\
+                --swiftc="/swift/usr/bin/swiftc" \\
+                --build-dir="/tmp/XCTest_build" \\
+                --foundation-build-dir "/swift/usr/lib/swift/linux" \\
+                --library-install-path="/swift/usr/lib/swift/linux" \\
+                --module-install-path="/swift/usr/lib/swift/linux/x86_64"
+            """))
     subparsers = parser.add_subparsers(
-        description="Use one of these to specify whether to build, test, "
-                    "or install XCTest. If you don't specify any of these, "
-                    "'build' is executed as a default. You may also use "
-                    "'build' to also test and install the built products. "
-                    "Pass the -h or --help option to any of the subcommands "
-                    "for more information.")
+        description=textwrap.dedent("""
+            Use one of these to specify whether to build, test, or install
+            XCTest. If you don't specify any of these, 'build' is executed as a
+            default. You may also use 'build' to also test and install the
+            built products. Pass the -h or --help option to any of the
+            subcommands for more information."""))
 
     build_parser = subparsers.add_parser(
         "build",
-        description="Build XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc "
-                    "using the given Swift compiler. This command may also "
-                    "test and install the built products.")
+        description=textwrap.dedent("""
+            Build XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc using the
+            given Swift compiler. This command may also test and install the
+            built products."""))
     build_parser.set_defaults(func=_build)
     build_parser.add_argument(
         "--swiftc",
         help="Path to the 'swiftc' compiler that will be used to build "
              "XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc. This will "
              "also be used to build the tests for those built products if the "
              "--test option is specified.",
-        metavar="PATH",
```

Comment:
I didn't think they added any value, and I figured I'd just include them in this pull request.

This is the help text diff:

```diff
diff --git a/build_script_help_text.txt b/build_script_help_text.txt
index 194df0d..e28cd3b 100755
--- a/build_script_help_text.txt
+++ b/build_script_help_text.txt

$ build_script.py build -h
# ...
optional arguments:
  -h, --help            show this help message and exit
-  --swiftc PATH       Path to the 'swiftc' compiler that will be used to
+  --swiftc SWIFTC     Path to the 'swiftc' compiler that will be used to
                        build XCTest.so, XCTest.swiftmodule, and
                        XCTest.swiftdoc. This will also be used to build the
                        tests for those built products if the --test option is
                        specified.
-  --build-dir PATH
+  --build-dir BUILD_DIR
                        Path to the output build directory. If not specified,
                        a temporary directory is used
-  --foundation-build-dir PATH
+  --foundation-build-dir FOUNDATION_BUILD_DIR
                        Path to swift-corelibs-foundation build products,
                        which the built XCTest.so will be linked against.
-  --swift-build-dir PATH
+  --swift-build-dir SWIFT_BUILD_DIR
                        deprecated, do not use
  --arch ARCH           deprecated, do not use
-  --module-install-path PATH
+  --module-install-path MODULE_PATH
                        Location at which to install XCTest.swiftmodule and
                        XCTest.swiftdoc. This directory will be created if it
                        doesn't already exist.
-  --library-install-path PATH
+  --library-install-path LIB_PATH
                        Location at which to install XCTest.so. This directory
                        will be created if it doesn't already exist.
  --release             builds for release
  --debug               builds for debug (the default)
  --test                Whether to run tests after building. Note that you
                        must have cloned https://github.com/apple/swift-llvm
                        at /Users/bgesiak/GitHub/apple/llvm in order to run
                        this command.
```

Personally I think it's less code without them, and removing them provides clearer help text. What do you think?